### ISSUE TITLE
tokenmeta#MetadataDeserialize returns strings with nul bytes

### ISSUE THREAD

**Issue by omarkilani** [portto/solana-go-sdk#20]
For whatever reason, all strings returned by MetadataDeserialize have \0 bytes padding them at the end.

It's hard to see it if you use fmt %v because they're ignored there, but if you look at the bytes of the string you'll see them.

**Comment by yihau** [comment]
Actually, all the bytes (including null) are the original data on chain.
I want to preserve them in the underlayer function.
WDYT?

**Comment by omarkilani** [comment]
Thanks for your response!

It's very confusing because the metadata Uri, when used with net/http, will always return an error due to null padding.

If the metadata name is passing to some formatting function then it will end up including the null bytes that you can't see.

IMHO, if the code is returning a string, it should be as expected by Go, i.e. excluding nulls.

If it was returning []byte then I think keeping the data like what's on the chain makes sense.

Right now, any time I'm working with metadata I need to use something like this:

```
func ByteSliceToString(s []byte) string {
        if i := bytes.IndexByte(s, 0); i != -1 {
                s = s[:i]
        }
        return string(s)
}

func CleanString(x string) string {
        return ByteSliceToString([]byte(x))
}
```

Which makes it cumbersome to work with the API, IMHO.

**Comment by yihau** [comment]
yup. I agree with you.
Absolutely it is clumsy to do the `CleanString`
Thank for the bug report. I will try to figure out a way to deal it.

**Comment by omarkilani** [comment]
FWIW, this is an example of the metadata serialised into JSON to see the null bytes

```
{"Key":4,"UpdateAuthority":"AumNBDNhvvknpcK9Zg6YeX9FrEoXtDE1f4rhK4qfVoyp","Mint":"2FREdEicgDaV4qixzuoSaJGY49SM5u4S8oRDK2MWVw8r","Data":{"Name":"Trippy Bunny #669\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","Symbol":"\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","Uri":"https://arweave.net/SfdF4FtlF6C8Ex_6V_D6JiF227jF0xCqMDa52lKN-UM\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000","SellerFeeBasisPoints":420,"Creators":[{"Address":"BVpxLszd8FLUd7N8trW2Ykq47PNHEojMpEu2qqy9KX1S","Verified":true,"Share":0},{"Address":"GRUZBgdF7vHj52bca5wpUBaJ4VtC3v6iyHLyJ5SKE4mD","Verified":false,"Share":25},{"Address":"2rwAUSFw9ZffXra6uJenBbMmKKsowzkgkhaVaQAPTDEH","Verified":false,"Share":25},{"Address":"A1PYqVforDNbobBJw4MY4jcPWnKFzCA5XpJTNhMdVyy8","Verified":false,"Share":25},{"Address":"4xjqkXXh4FcirWZuFFJZypKy5TU36SFwrF6UFXos2Lii","Verified":false,"Share":25}]},"PrimarySaleHappened":true,"IsMutable":true,"EditionNonce":249}
```

**Comment by yihau** [comment]
I've released a new version, v1.8.1
I trim the null byte in the string.
Feel free to open another issue for any bugs!
